### DISCUSSION TITLE
Accumulated settings - are .vscode/tasks.json able to be merged by user's custom settings?

### DISCUSSION THREAD

**Discussion by PureKrome** [microsoft/vscode-discussions#2415] [Extension Development QnA] [answered]
Does VS Code allow developers to have their own task settings which override any 'repo' tasks.json settings?

for example:

```
|_Projects
  |__FancyPantsRepo
    |_.vscode (project 'vs code settings')
      |_tasks.json
      |_settings.json
    |_ (the rest of your repo's files)
  |_.vscode
    |_tasks.json (my custom tasks)
```

So with this repo. we start with the lowest level (repo -> .vscode -> tasks.json) and then 'go upwards' till we stop. One level above, we find _another_ .vscode folder! in that is tasks.json and it's _merged_ with the repo's tasks.json

- new task, it's "added"
- existing task, it's "replaced"

To determine what is existing or new, we could use the `label` property of an individual task, in the `tasks` array.

Does this exist? If not, I'm happy to propose the existing feature.

**Comment by isidorn** [comment, marked as answer]
This discussion repo is for VS Code extension authors, not for users. For that question please try out on stack overflow.
In case you are writing a VS Code extensions do tell us what you are trying to do, and what are the issues.

Having said that, this should be possible. Please read our docs https://code.visualstudio.com/docs/editor/tasks
Or ask Copilot

**Reply by PureKrome** [reply]
Thanks kindly @isidorn for the reply - appreciate it :) Also apologies to posting this in the wrong forum. I can't use SO because that's a site for Q/A and this is really a _discussion_. Also, I failed to actually find any official VSCode _discussions_ which are reviewed by VSCode maintainers, etc. :(

I'll check the docs you've provided. I think i didn't have much luck with CP but that could have been a skill issue with me + my prompting.

Thanks again for the help - my search continues! 🥾🚶🏻‍♂️‍➡️
